### PR TITLE
Feat: Transaction Return to Analyst - 382

### PR DESCRIPTION
**Pull request by AlexZorkin**
- Added new button config option for return to analyst button on transaction view
- added tests for post action navigation

### CODE REVIEW

**Review by hamed-valiollahi — APPROVED** [review]
Code looks great! Thanks, Alex.

I noticed a few things after testing it locally that I wanted to share with you:

1. The "Return to analyst" button is visible to the analyst themselves, allowing them to return the transaction to themselves.

2. When a transaction is returned to the analyst, its status changes from "Recommended" to "Draft." It would be helpful to have a new status or a boolean flag to distinguish between these states.

3. The transaction history becomes a bit confusing after returning the transaction twice. Please see the attached screenshot.

4. When I tried to create an "Administrative adjustment" transaction, it was saved as an "Initiative agreement." I recreated the containers and database, but the issue persists.

Thank you!

Screen Shot 2024-05-22 at 10 48 46 AM

**Comment by AlexZorkin** [comment]
Thanks Hamed, I will address these issues.

**Comment by AlexZorkin** [comment]
- fixed button visibility, only director can return txn now
- added a returned bool to the response after a txn is returned, in order to handle redirect logic
- clean up txn history logic. Now will update a history record if it already exists rather than duplicating.
- fixed form issue on radio btn selection

File: frontend/src/views/Transactions/AddEditViewTransaction.jsx [hunk 1]
```diff
@@ -54,82 +53,94 @@ export const AddEditViewTransaction = () => {
   const mode = matches[matches.length - 1]?.handle?.mode
   const { transactionId } = useParams()
   const { data: currentUser, hasRoles, hasAnyRole } = useCurrentUser()
-  const isGovernmentUser = currentUser?.isGovernmentUser
   const [steps, setSteps] = useState(['Draft', 'Recommended', 'Approved'])
   const alertRef = useRef()
   const [alertMessage, setAlertMessage] = useState('')
   const [alertSeverity, setAlertSeverity] = useState('info')
+<<<<<<< feat/alex-return-txn-to-analyst-382
```
Comment (hamed-valiollahi) [inline frontend/src/views/Transactions/AddEditViewTransaction.jsx:60]: Could you please resolve the conflicts? There are unresolved merge conflicts in the branch that need to be fixed before I can proceed with testing. Thank you

**Comment by hamed-valiollahi** [comment]
@AlexZorkin Hi Alex, I'm getting a 'govRoles is not defined' error when I click on the 'New Transaction' button.

**Pull request merged**